src/main/java/taxi/dao/DriverDaoImpl.java

```diff
@@ -101,12 +103,24 @@ public boolean delete(Long id) {
         }
     }
 
+    @Override
+    public Optional<Driver> findByLogin(String login) {
```

Comment:
It’s not a good solution to return all data from DB and filter it on the API side/ You should create a query that returns only one record.